### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version preview/2022-04-30-preview

### PR DESCRIPTION
**Pull request by didact2401**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/didact2401/azure-rest-api-specs/dev-iothub-Microsoft.Devices-2022-04-30-preview...main/).

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that label "WaitForARMFeedback" will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by rkmanda** [comment]
>         },

x-ms-secret?

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:3220 in c0ba99e. [](commit_id = c0ba99ecc6df5d3ed7f518532f75d54befd637f7, deletion_comment = False)

**Comment by rkmanda** [comment]
>         },

x-ms-secret?

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:3224 in c0ba99e. [](commit_id = c0ba99ecc6df5d3ed7f518532f75d54befd637f7, deletion_comment = False)

**Comment by rkmanda** [comment]
>         },

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

---
In reply to: [1125583794](https://github.com/Azure/azure-rest-api-specs/pull/18743#issuecomment-1125583794) [](http://example.com/codeflow?ancestors=1125583794)

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:3220 in 462851d. [](commit_id = 462851d4197aed39d380e33101344cde7e6d9d51, deletion_comment = False)

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

**Review by rkmanda — APPROVED** [review]
:shipit:

**Comment by jorgecotillo** [comment]
>           "x-ms-secret": true

Just so you are aware, marking a property as x-ms-secret it means you cannot return this property in a GET call, retrieving secrets from a resource is meant to be a POST action, see "secrets" from here: [https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review](https://armwiki.azurewebsites.net/rp_onboarding/process/api_review_best_practices.html#common-issues-found-in-review)

---
Refers to: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:3001 in 3b1b3a0. [](commit_id = 3b1b3a0797c70044a48712d8ac068a5250f58ebf, deletion_comment = False)

**Comment by jorgecotillo** [comment]
> > 
> 
> Ack'd.

Good, then make sure to remove your secrets from your new examples please.

**Review by vikramadityab72 — APPROVED** [review]

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json [hunk 1]
```diff
@@ -0,0 +1,4637 @@
...
+            "name": "If-Match",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "description": "ETag of the IoT Hub. Do not specify for creating a brand new IoT Hub. Required to update an existing IoT Hub."
+          }
+        ],
+        "responses": {
+          "201": {
+            "headers": {
```
Comment (raych1) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:164]: This should be the definition of header instead of the value.
Pls refer to https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md#headers-object.

**Comment by jianyexi** [comment]
Noticed that you have added a suppression in the readme.md, please follow the https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process to get approval

**Comment by weidongxu-microsoft** [comment]
@tadelesh  on Go breaking

.net can be ignored.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json [hunk 2]
```diff
@@ -2932,7 +2972,8 @@
         },
         "connectionString": {
           "description": "The connection string of the service bus queue endpoint.",
-          "type": "string"
+          "type": "string",
+          "x-ms-secret": true
```
Comment (weidongxu-microsoft) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:2976]: Does it mean the property will not be returned in response?

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json [hunk 3]
```diff
@@ -4239,7 +4358,7 @@
           "type": "string"
         },
         "inputBlobName": {
-          "description": "The blob name to be used when importing from the provided input blob container.",
+          "description": "from the provided input blob container.",
```
Comment (weidongxu-microsoft) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:4361]: why the change on description? seems the original one is better?

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json [hunk 4]
```diff
@@ -4452,6 +4574,22 @@
...
       }
+    },
+    "RootCertificateProperties": {
+      "description": "This property store root certificate related information",
+      "type": "object",
+      "properties": {
+        "enableRootCertificateV2": {
+          "description": "This property when set to true, hub will use G2 cert; while it's set to false, hub uses Baltimore Cert.",
+          "type": "boolean"
+        },
```
Comment (weidongxu-microsoft) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:4585]: Should this better be an enum (if you later have more than G2 and Baltimore)?

**Review by weidongxu-microsoft — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
A few comments. LGTM.

Pending on approval from Go SDK.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json (hunk 3 above)
Comment (firstaye-zz) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:4361]: Good catch, it will be reverted.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json (hunk 4 above)
Comment (firstaye-zz) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:4585]: No, the feature will only be in preview version, we will retired the API soon after the deadline of Baltimore cert sunset.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json (hunk 2 above)
Comment (firstaye-zz) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:2976]: I believe so, @didact2401 would you double check on this?

Comment (didact2401) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2022-04-30-preview/iothub.json:2976]: I believe that is correct, yes.

**Review by weidongxu-microsoft — APPROVED** [review]
